### PR TITLE
Dcallaghan/mediatag embed #158

### PR DESCRIPTION
**Pull request by xiaoniuniu89**
# ✨ Codu Pull Request 💻

Codu Logo

👉 _**Please remove the below and replace with your own values, leaving the headers where they are.**_ 👈

## Pull Request details:
Updates custom tags to work in one general media tag, user can now use {% media src="src-to-embed" /%} for youtube, codepen, code sandbox and will fallback to generic iframe if not one of those types. Add some checking for youtube incase user copy link from url bar and not from embed link. both should work now.

## Any Breaking changes:
- none, user can still use old tags or this new tag, both will work

## Associated Screenshots:
    
can manually test with:

Youtube

{% media src="https://www.youtube.com/watch?v=7mmlmxamw_k" /%}

{% media src="www.youtube.com/watch?v=7mmlmxamw_k" /%}

{% media src="https://youtu.be/7mmlmxamw_k" /%}

{% media src="youtu.be/7mmlmxamw_k" /%}

Code Sandbox
{% media src="https://codesandbox.io/embed/vanilla-vanilla?fontsize=14&hidenavigation=1&theme=dark" / %}

{% media src="https://codesandbox.io/s/vanilla-vanilla" /%}

Codepen

{% codepen src="https://codepen.io/yuanchuan/pen/dyqYYBN" /%}

{% media src="codepen.io/yuanchuan/pen/dyqYYBN" /%}

Fallback

{% media src="https://www.codu.co/" /%}
{% media src="www.codu.co/" / %}

### CODE REVIEW

File: components/markdocNodes/Media/Media.tsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  // format youtube url if it is not embed format
+  // can happen when user copy and past direct from youtube url bar
+  const formatEmbed = (src: string) => {
+    const videoId = src.match(/(?:\?v=|&v=|youtu\.be\/)([\w-]{11})/)[1] || '';
+    src = `https://www.youtube.com/embed/${videoId}`;
+    return src;
+  };
+
+  const processedProps = processMedia(props);
+  console.log(processedProps);
```
Comment (NiallJoeMaher) [inline components/markdocNodes/Media/Media.tsx]: I don't think we need this log. 😁

**Review by NiallJoeMaher — CHANGES_REQUESTED** [review]
Tiny change! Just get rid of the `console.log`. Great work!

File: components/markdocNodes/Media/Media.tsx (hunk 1 above)
Comment (xiaoniuniu89) [reply components/markdocNodes/Media/Media.tsx]: np, I missed it.

**Review by NiallJoeMaher — APPROVED** [review]
Another great addition! Looks like it will be easy to expand too. 🦾 

Thanks!

**Review by NiallJoeMaher — CHANGES_REQUESTED** [review]
Ooof I spoke too soon! The build failed 😢 

Seems to be an error importing the old elements into the Media block: 
```

[21:22:14.585] Running build in Cleveland, USA (East) – cle1
[21:22:14.633] Cloning github.com/codu-code/codu (Branch: dcallaghan/MediatagEmbed-#158, Commit: 9d220e2)
[21:22:14.641] Skipping build cache, deployment was triggered without cache.
[21:22:15.109] Cloning completed: 475.531ms
[21:22:15.289] Running "vercel build"
[21:22:15.869] Vercel CLI 28.16.12
[21:22:16.233] Installing dependencies...
[21:22:21.509] npm WARN deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
[21:22:39.802] 
[21:22:39.803] > codu@0.1.0 postinstall
[21:22:39.803] > prisma generate || true
[21:22:39.803] 
[21:22:41.483] Prisma schema loaded from prisma/schema.prisma
[21:22:42.564] 
[21:22:42.564] ✔ Generated Prisma Client (4.10.1 | library) to ./node_modules/@prisma/client in 351ms
[21:22:42.564] You can now start using Prisma Client in your code. Reference: https://pris.ly/d/client
[21:22:42.564] ```
[21:22:42.565] import { PrismaClient } from '@prisma/client'
[21:22:42.565] const prisma = new PrismaClient()
[21:22:42.565] ```
[21:22:42.637] 
[21:22:42.637] added 739 packages in 26s
[21:22:42.638] 
[21:22:42.638] 130 packages are looking for funding
[21:22:42.638]   run `npm fund` for details
[21:22:42.663] Detected Next.js version: 13.1.1
[21:22:42.679] Detected `package-lock.json` generated by npm 7+...
[21:22:42.680] Running "npm run build"
[21:22:43.110] 
[21:22:43.110] > codu@0.1.0 build
[21:22:43.110] > next build
[21:22:43.111] 
[21:22:43.642] warn  - You have enabled experimental feature (swcPlugins) in next.config.js.
[21:22:43.643] warn  - Experimental features are not covered by semver, and may cause unexpected or broken application behavior. Use at your own risk.
[21:22:43.643] 
[21:22:43.664] Attention: Next.js now collects completely anonymous telemetry regarding usage.
[21:22:43.664] This information is used to shape Next.js' roadmap and prioritize features.
[21:22:43.665] You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
[21:22:43.665] https://nextjs.org/telemetry
[21:22:43.665] 
[21:22:43.802] info  - Skipping validation of types
[21:22:43.802] info  - Linting...
[21:22:51.896] 
[21:22:51.900] ./pages/articles/index.tsx
[21:22:51.900] 26:6  Warning: React Hook useEffect has missing dependencies: 'fetchNextPage' and 'hasNextPage'. Either include them or remove the dependency array.  react-hooks/exhaustive-deps
[21:22:51.901] 
[21:22:51.901] ./pages/create/[[...postIdArr]].tsx
[21:22:51.902] 205:6  Warning: React Hook useEffect has a missing dependency: 'reset'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps
[21:22:51.902] 212:6  Warning: React Hook useEffect has missing dependencies: 'allowUpdate', 'body', 'data?.body', 'data?.title', 'published', 'savePost', and 'title'. Either include them or remove the dependency array.  react-hooks/exhaustive-deps
[21:22:51.902] 217:6  Warning: React Hook useEffect has a missing dependency: 'router'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps
[21:22:51.903] 226:6  Warning: React Hook useEffect has a missing dependency: 'isDirty'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps
[21:22:51.903] 
[21:22:51.903] ./pages/notifications/index.tsx
[21:22:51.904] 16:1  Warning: All imports in the declaration are only used as types. Use `import type`.  @typescript-eslint/consistent-type-imports
[21:22:51.904] 56:6  Warning: React Hook useEffect has missing dependencies: 'fetchNextPage' and 'hasNextPage'. Either include them or remove the dependency array.  react-hooks/exhaustive-deps
[21:22:51.905] 
[21:22:51.908] ./components/Comments/CommentsArea.tsx
[21:22:51.909] 114:6  Warning: React Hook useEffect has a missing dependency: 'initiallyLoaded'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps
[21:22:51.909] 
[21:22:51.909] ./components/Footer/Footer.tsx
[21:22:51.910] 1:15  Warning: 'NextPage' is defined but never used.  @typescript-eslint/no-unused-vars
[21:22:51.910] 
[21:22:51.910] ./components/PromptService/PromptService.tsx
[21:22:51.911] 9:27  Warning: Unexpected any. Specify a different type.  @typescript-eslint/no-explicit-any
[21:22:51.911] 9:37  Warning: Unexpected any. Specify a different type.  @typescript-eslint/no-explicit-any
[21:22:51.912] 35:5  Warning: React Hook useCallback has missing dependencies: 'router.events' and 'updateParent'. Either include them or remove the dependency array. If 'updateParent' changes too often, find the parent component that defines it and wrap that definition in useCallback.  react-hooks/exhaustive-deps
[21:22:51.912] 58:6  Warning: React Hook useEffect has missing dependencies: 'removeListener' and 'router.events'. Either include them or remove the dependency array.  react-hooks/exhaustive-deps
[21:22:51.912] 
[21:22:51.913] ./components/markdocNodes/Media/Media.tsx
[21:22:51.913] 99:54  Warning: Unexpected any. Specify a different type.  @typescript-eslint/no-explicit-any
[21:22:51.913] 
[21:22:51.914] info  - Need to disable some ESLint rules? Learn more here: https://nextjs.org/docs/basic-features/eslint#disabling-rules
[21:22:51.916] info  - Creating an optimized production build...
[21:23:07.768] Failed to compile.
[21:23:07.768] 
[21:23:07.768] ./components/markdocNodes/Media/Media.tsx
[21:23:07.768] Module not found: Can't resolve '../Codepen/CodePen' in '/vercel/path0/components/markdocNodes/Media'
[21:23:07.768] 
[21:23:07.769] Import trace for requested module:
[21:23:07.769] ./components/markdocNodes/Media/Media.tsx
[21:23:07.772] ./markdoc/components.ts
[21:23:07.772] 
[21:23:07.772] ./components/markdocNodes/Media/Media.tsx
[21:23:07.773] Module not found: Can't resolve '../Codesandbox/CodeSandbox' in '/vercel/path0/components/markdocNodes/Media'
[21:23:07.773] 
[21:23:07.773] Import trace for requested module:
[21:23:07.773] ./components/markdocNodes/Media/Media.tsx
[21:23:07.773] ./markdoc/components.ts
[21:23:07.773] 
[21:23:07.774] 
[21:23:07.774] > Build failed because of webpack errors
[21:23:07.822] Error: Command "npm run build" exited with 1
```

**Comment by xiaoniuniu89** [comment]
should work now with last commit, if not let me know!

**Review by NiallJoeMaher — APPROVED** [review]

**Pull request merged**